src/ui/dashboardpage.cpp

```diff
@@ -209,31 +209,71 @@ Action getAction(bool frameHovered, bool hoveredInTitleArea, const ImVec2 &scree
     }
     return finalAction;
 }
+} // namespace
 
-constexpr int gridSizeW = 16;
-constexpr int gridSizeH = 16;
 
-} // namespace
 
-void alignForWidth(float width, float alignment = 0.5f) noexcept {
+static bool plotButton(App *app, std::string_view glyph, std::string_view tooltip) noexcept {
+    ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleColor(ImGuiCol_ButtonHovered, ImVec4(0, 0, 0, 0.1f));
+    ImGui::PushStyleColor(ImGuiCol_ButtonActive, ImVec4(0, 0, 0, 0.2f));
+    ImGui::PushFont(app->fontIconsSolid);
+    const bool ret = ImGui::Button(glyph.data());
+    ImGui::PopFont();
+    ImGui::PopStyleColor(3);
+    if (ImGui::IsItemHovered()) {
+        ImGui::SetTooltip("%s", tooltip.data());
```

Comment:
SetTooltip() expects the string to be null terminated, tooltip.data() does not guarantee that.  It's true that all the calls to plotButton() pass a string literal so in practice this is not a problem now, but in order to prevent possible mistakes in the future it may be a good idea to change the argument type to const char*.

Reply:
Or another option would be to use instead Begin/EndTooltip() and the TextUnformatted() overload that takes a begin and end pointer.

Reply:
or `fmt::format(...).c_str()'

Reply:
or just std::string